### ISSUE TITLE
Reality model loading issue

### ISSUE THREAD

**Issue by dontbmh** [iTwin/itwinjs-core#2035]
Hi,

we got a reality model dataset exported through a third-party tool in 3D tileset format. Then the tilesets are loaded via displayStyle.attachRealityModel but there is nothing present. After inspecting the network panel from browser devtools, it can be seen that the tilesets(b3dm) and their corresponding images(blob, image/jpeg) are requested successfully while locating to the area with the reality model attached and there is no logs/errors from imodeljs. 

Is imodeljs fully compatible with 3D tilesets? If not, are there any specs/settings that need to be taken care of when generating the 3D tilesets?

For the test purpose, these are two temporary URLs given for Cesium viewer and reality model 3D Tileset, respectively.
http://20.48.110.217/reality/cesium1.html
http://20.48.110.217/reality/tilesetjson/tileset.json

**Comment by AlainRobertAtBentley** [comment]
@dontbmh Just answered something concerning elevation on the issue you just closed. The team in charge of Reality Data and geolocation just happens to be the same. As I indicated it would probably be easier for you to open a Service Request for MicroStation. One of the service technician will be assigned and he will keep you updated more effectively than we can here.
We did notice that horizontal positioning appeared correct but could not detemine the source of the 15 meter elevation shift. We will keep you updated in any case.

**Comment by rbois-bentley** [comment]
@dontbmh You could try to disable Draco compression and if that still doesn't work you could try exporting to 3DTiles using GLTF version 1.0 if you can. Not sure if Draco + GLTF 2.0 are supported yet in iModeljs.

**Comment by pmconne** [comment]
glTF 2.0 is supported.
We have an [implementation of Draco mesh decoding](https://github.com/imodeljs/imodeljs/blob/master/core/frontend/src/tile/DracoDecoder.ts) but it is [not fully hooked up](https://github.com/imodeljs/imodeljs/blob/59cde824a362ed0273176856c167fef1ea0cd88e/core/frontend/src/tile/GltfReader.ts#L559).
@RBBentley any reason not to add the Draco package dependency and enable decoding?

**Comment by RBBentley** [comment]
Yes -- This tileset uses Draco3D which as @pmconne points out is currently disabled.

@dontbmh Was this tileset created with the Cesium pipeline? -- Can you retest with Draco compression disabled?

**Comment by dontbmh** [comment]
@RBBentley Actually,  we are using a third-party tool to export the tilesets with glTF 2.0 and Draco compression disabled for sure. After an initial investigation, it can be confirmed that this tool is developed based on Ceisum things but the versions of those are still unspecified yet. Moreover, the lack of error/logging outputs makes the debug more complicated but we are continuously trying.

**Comment by pmconne** [comment]
The tileset definitely uses Draco compression as revealed in debugger.
[GltfReader.readMeshPrimitive](https://github.com/iTwin/itwinjs-core/blob/fa636a0c7e3eab21a8fc15b6ec86c0400285f8ed/core/frontend/src/tile/GltfReader.ts#L1341) returns undefined here:

```
    if (primitive.extensions?.KHR_draco_mesh_compression) {
      return undefined; // Defer Draco decompression until web workers implementation.
```
![image](https://user-images.githubusercontent.com/22944042/150409568-edc2f9ed-c13a-4951-b58a-d4ea9cb92822.png)

**Comment by pmconne** [comment]
@dontbmh I have implemented support for draco mesh decoding in #3106. If you want me to test it against your reality model, please supply a working version. The one previously supplied produces URLs that go outside the root directory - e.g., "http://localhost:8080/dontbmh/../Tile_0003_0020/Tile_0003_0020.b3dm". Otherwise, I will close this issue after the PR merges.
